torch/csrc/jit/script/logging.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+namespace torch {
+namespace jit {
+namespace logging {
+
+class LoggerBase {
+ public:
+  TORCH_API virtual void addStatValue(
+      const std::string& stat_name,
+      int64_t val) = 0;
+  TORCH_API virtual int64_t getCounterValue(const std::string& name) const = 0;
```

Comment:
What is the expected contract here? At this level it is abstract what getCounterValue returns. In the example it is aggregated value, but the aggregation type is set per-stat by a subclass. It is weird that half the API is in the abstract class.